### PR TITLE
Dashboard + Checklist + Cohesiveness 

### LINKED ISSUE #56: Dashboard
A tiered dashboard for guided setup. The most important/baseline items are at the top, and optional/later items are towards the bottom. Items in lower tiers should be grayed out, yet I don't think they should be disabled (just discouraged).

Dashboard cards will link out to either a Checklist, Wizard, or regular link. The easiest implementation would be to just make the cards a regular link, and we can link to anything.

Render the dashboard in the general Newspack admin page that currently has "TODO: A dashboard page here or something." as its content.

Design:
1  Newspack setup calypsoified

### PR DESCRIPTION
**Pull request by claudiulodro**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #56 .

Sorry for YAGPR (Yet Another Giganto Pull Request 😄). The components are somewhat woven together, and I didn't feel that the architecture could be properly evaluated or communicated if it was split into a bunch of smaller PRs.

This pull request implements dashboard and checklist handling and ties everything together into one (mostly) functional application.

Architecture:
- The Dashboard is a stand-alone area that can render anything into a card (checklists, links, etc.). Under the hood, it's just a 1-screen wizard. See `assets/wizards/dashboard` and `includes/wizards/class-dashboard.php`.
- Checklists are managed and rendered in a centralized way, so we don't need to make a bunch of different independent checklist handlers. Checklists are defined and routed in  `includes/class-checklists.php`. Checklists are rendered in `assets/wizards/checklist`.
- Wizards contain all of the info in their classes (name, description, etc.). They are managed and initialized in `includes/class-wizards.php`. The `Wizards` class provides a central location for retrieving wizards and their info.

Screenshots in details:
<details>

Screen Shot 2019-06-10 at 11 20 32 AM
Screen Shot 2019-06-10 at 11 20 40 AM
Screen Shot 2019-06-10 at 11 21 00 AM
Screen Shot 2019-06-10 at 11 21 14 AM

</details>

### How to test the changes in this Pull Request:

1. `npm ci; npm run clean; npm run build:webpack`
2. Go to WP Admin > Newspack. Observe most wizards don't have links in the submenu any more.
3. On the dashboard, observe things look good, with a mix of statuses on the cards.
4. Click the "Demo Checklist" or "Memberships" cards. Observe you are taken to the appropriate checklist.
5. Click the "Skip" in the checklist to skip an item or "Do it" to go to the appropriate wizard. Click "I'm done setting up" to return to the dashboard.

The main thing not done in this PR is making the checklist item completion permanent; that'll be done in a future PR.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by claudiulodro** [comment]
Probably don't want to merge this until #89 is ready, so we're sure it all works together nicely.

File: includes/wizards/class-subscriptions-wizard.php [hunk 1]
```diff
@@ -49,6 +42,33 @@ public function __construct() {
 		add_filter( 'woocommerce_product_data_store_cpt_get_products_query', [ $this, 'handle_only_get_newspack_subscriptions_query' ], 10, 2 );
 	}
 
+	/**
+	 * Get the name for this wizard.
+	 *
+	 * @return string The wizard name.
+	 */
+	public function get_name() {
+		return __( 'Subscriptions', 'newspack' );
```
Comment (philipjohn) [inline includes/wizards/class-subscriptions-wizard.php]: We should always use the escaping alternatives for localisation functions where possible to ensure that translations don't introduce any unsafe output. E.g. `esc_html__()`

File: includes/wizards/class-wizard.php [hunk 2]
```diff
@@ -87,4 +91,58 @@ public function api_permissions_check( $request ) {
...
+		}
+	}
+
+	/**
+	 * Get the URL for this wizard.
+	 *
+	 * @return string
+	 */
+	public function get_url() {
+		return admin_url( 'admin.php?page=' . $this->slug );
```
Comment (philipjohn) [inline includes/wizards/class-wizard.php]: Let's make sure this is safe by wrapping it in `esc_url()` too

File: includes/class-checklists.php [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+	 */
+	public static function get_status( $checklist_slug ) {
+		return 'enabled';
+	}
+
+	/**
+	 * Register and add the page that the checklists will live on.
+	 */
+	public static function add_page() {
+		add_submenu_page( 'newspack', 'Checklist', 'Checklist', 'manage_options', 'newspack-checklist', 'Newspack\Checklists::render_checklist' );
```
[the remark this answers is not in the document]
Comment (claudiulodro) [reply includes/class-checklists.php]: It still needs to be an admin page; we just don't want a link to it to show up in the admin sidebar. Does that answer your question? I'm not super sure exactly what you're saying here. :)

File: includes/class-checklists.php [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+				'length'      => $wizard->get_length(),
+				'completed'   => $wizard->is_completed(),
+			];
+		}
+
+		wp_register_script(
+			'newspack-checklist',
+			Newspack::plugin_url() . '/assets/dist/checklist.js',
+			[ 'wp-components' ],
+			filemtime( dirname( NEWSPACK_PLUGIN_FILE ) . '/assets/dist/checklist.js' ),
```
[the remark this answers is not in the document]
Comment (claudiulodro) [reply includes/class-checklists.php]: For sure, I mostly didn't want the overhead of managing versions and having to make sure caches are busted while we're developing the plugin. Is it cool with you if we wait until the plugin is closer to release before making that change?

**Comment by claudiulodro** [comment]
Thanks for the feedback, @philipjohn! I'll get your requested changes made before merging.

> One thing I did encounter, which may be entirely unrelated to this change was that going through the subscriptions wizard I noticed that once I'd added a subscription there was no way to get back to the checklist without going back through the dashboard.

Yep; that needs to be added to any existing wizards. It's definitely part of the original designs, and was omitted because there was nowhere for the link to point to until now. I've created an issue around this at https://github.com/Automattic/newspack-plugin/issues/98

**Comment by claudiulodro** [comment]
Requested changes made (See commit log if interested in specifics):
- Removed unused `init_hooks`.
- Added some php comments around the localized JS variables.
- Added escaping where needed.

File: includes/class-checklists.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (philipjohn) [reply includes/class-checklists.php]: Oh, does WP need use to call `add_submenu_page()` in order for the page to be accessible at all, even if we hide it from the menu? Perhaps that's the nuance I was missing in my understanding.

File: includes/class-checklists.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (philipjohn) [reply includes/class-checklists.php]: Definitely! I've created #99 so we don't forget :)

File: includes/class-checklists.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (claudiulodro) [reply includes/class-checklists.php]: Yes, exactly! There's all sorts of permissions handling, routing, etc. that `add_submenu_page` handles. To test you can try going to a non-existent page e.g. `http://local.newspack.test/wp-admin/admin.php?page=blah` and see you're prevented from accessing it.

**Pull request merged**